### PR TITLE
Docs/recipes sourcing from yaml json

### PR DESCRIPTION
**Pull request by jonniebigodes**
## Description
First draft of the recipe for importing json and yaml directly into a page/component.

Created a simple and concise example on how to consume both types and the category called "Sourcing Content" where this recipe and any others will reside.

## Related Issues
Addresses #14976

cc @marcysutton

### CODE REVIEW

File: docs/docs/recipes.md [hunk 1]
```diff
@@ -226,6 +227,46 @@ Data sourcing in Gatsby is plugin-driven; Source plugins fetch data from their s
...
+import YAMLContent from "your-folder/your-yaml-file.yaml"
+
+export default () => (
+  <div style={{ color: `purple` }}>
+    <p>YAMLContent.the-property-you-want-to-display</p>
+  </div>
+)
+```
+
+2. To import JSON content directly into a page, open the page-2.js page component (src/pages/page-2.js) import the JSON file into the component as a default export.
```
Comment (marcysutton) [inline docs/docs/recipes.md]:
```suggestion
#### JSON Directions

1. Open the component in `src/pages/index.js`, import the JSON file, and assign it to a variable.

2. In the JSX markup, reference the variable and data structure returned from the JSON file to display it.
```
By listing the two options as separate and optional steps, we can eliminate the `page-2` requirement.

File: docs/docs/recipes.md [hunk 2]
```diff
@@ -226,6 +227,46 @@ Data sourcing in Gatsby is plugin-driven; Source plugins fetch data from their s
 - Search available source plugins in the [Gatsby library](/plugins/?=source)
 - Understand source plugins by building one in the [Pixabay source plugin tutorial](/docs/pixabay-source-plugin-tutorial/)
 
+## Sourcing Content
+
+### Directly importing data (json & yaml)
+
+#### Prerequisites
+
+- A Gatsby site with two page components `index.js` and `page-2.js`
```
Comment (marcysutton) [inline docs/docs/recipes.md]:
```suggestion
- A [Gatsby site](/docs/quick-start/) with an `index.js` page component
```

See reasoning about one component, below!

File: docs/docs/recipes.md [hunk 3]
```diff
@@ -226,6 +227,46 @@ Data sourcing in Gatsby is plugin-driven; Source plugins fetch data from their s
...
+export default () => (
+  <div style={{ color: `purple` }}>
+    <p>YAMLContent.the-property-you-want-to-display</p>
+  </div>
+)
+```
+
+2. To import JSON content directly into a page, open the page-2.js page component (src/pages/page-2.js) import the JSON file into the component as a default export.
+
+```jsx:title=src/pages/page-2.js
```
Comment (marcysutton) [inline docs/docs/recipes.md]:
```suggestion
```jsx:title=src/pages/index.js
```

File: docs/docs/recipes.md [hunk 4]
```diff
@@ -226,6 +227,46 @@ Data sourcing in Gatsby is plugin-driven; Source plugins fetch data from their s
...
+import JSONContent from "your-folder/your-json-file.json"
+
+export default () => (
+  <div style={{ color: `purple` }}>
+    <p>JSONContent.the-property-you-want-to-display</p>
+  </div>
+)
+```
+
+3. Run `gatsby develop` and navigate to the index page. You should have a page with content sourced directly from a YAML file.
```
Comment (marcysutton) [inline docs/docs/recipes.md]:
```suggestion
3. Run `gatsby develop` and navigate to the index page. You should have a page with content sourced directly from a YAML or JSON file.
```

File: docs/docs/recipes.md [hunk 5]
```diff
@@ -226,6 +227,46 @@ Data sourcing in Gatsby is plugin-driven; Source plugins fetch data from their s
...
+   Navigate to page-2 and you should have a page with content sourced directly from a JSON file.
```
Comment (marcysutton) [inline docs/docs/recipes.md]:
```suggestion
```
By combining them, we can use one page and make it dependent on their use case.

**Comment by marcysutton** [comment]
As an FYI - this PR needs to be updated from master since the recipes page has evolved quite a bit in the past few days.

**Comment by jonniebigodes** [comment]
@marcysutton i've applied the changes you requested and with that it looks way more simple.

Now that i was able to see the massive overhaul done to the recipes document, congratulations on all parties involved. It now looks more streamlined. With this out of the way i would like to ask a couple of things:

1- Better to close this pull request and update my fork and create a new pull request with all of the changes?
2- Where would this recipe would be best suited? On item 5 `Sourcing data`?

**Comment by jonniebigodes** [comment]
closing this pr and update my fork so that the recipes doc is up to par with the latest changes to this document. Will open a new one soon

**Comment by sidharthachatterjee** [comment]
Thank you so much @jonniebigodes 

Cleaning up several old PRs at the moment so this is super useful! 🥇

**Pull request closed without merging**